runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkBatchPortablePipelineTranslator.java

```diff
@@ -418,4 +442,14 @@ private static String getOutputId(PTransformNode transformNode) {
   private static String getExecutableStageIntermediateId(PTransformNode transformNode) {
     return transformNode.getId();
   }
+
+  /** Predicate to determine whether a URN is a Spark native transform. */
+  @AutoService(NativeTransforms.IsNativeTransform.class)
+  public static class IsSparkNativeTransform implements NativeTransforms.IsNativeTransform {
```

Comment:
This isn't used, is it? I saw Flink does this too. But I don't get what is the issue with a service for this?

Reply:
I didn't dig into why, but the portable pipeline fuser complains unless we mark Reshuffle as a native (or "primitive") transform. https://github.com/apache/beam/blob/c565881b3041730f4e1206ed8404e4b0317e5037/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/QueryablePipeline.java#L231-L235

Reply:
Interesting, still curious about it, any ideas @angoenka or @mxm ?